src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,131 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnMarkCommand;
+
+/**
+ * Makes sense of the user's commands.
+ */
+public class Parser {
+    /**
+     * Returns corresponding command objects from user commands given in strings.
+     * @param str User command
+     * @return Command object corresponding to the user command.
+     * @throws RuntimeException If user commands don't make sense.
+     */
+    public static Command parse(String str) {
+        Command command;
+        boolean todoCheck = str.startsWith("todo ");
+        boolean deadlineCheck = str.startsWith("deadline ");
+        boolean eventCheck = str.startsWith("event ");
+        boolean deleteCheck = str.startsWith("delete ");
+        boolean markCheck = str.startsWith("mark ");
+        boolean unMarkCheck = str.startsWith("unmark ");
+        boolean listCheck = str.equals("list");
+        boolean findCheck = str.startsWith("find ");
+        boolean exitCheck = str.equals("bye");
+        boolean nothingCheck = str.equals("");
+        if (str.equals("todo") || str.equals("deadline") || str.equals("event") || str.equals("delete")
+                || str.equals("mark") || str.equals("unmark") || str.equals("find")) {
+            throw new RuntimeException("This command's field cannot be left blank!");
+        } else if (todoCheck || deadlineCheck || eventCheck) {
+            if (deadlineCheck) {
+                String target = " /by ";
+                if (!str.contains(target)) {
+                    throw new RuntimeException("Unable to create Deadline! Deadline commands need a /by field!");
+                }
+                int index = str.indexOf(target);
```

Comment:
this part is slightly harder to understand, maybe you can create a new method to handle dates.